plugins/TimingTriggerCandidateMaker.cpp

```diff
@@ -37,48 +38,87 @@ TimingTriggerCandidateMaker::TimingTriggerCandidateMaker(const std::string& name
+    uint32_t signal = 1 << bit_index;
+
+    if (!m_hsisignal_map.count(signal)) {
       throw dunedaq::trigger::SignalTypeError(ERS_HERE, get_name(), data.signal_map);
     }
-  }
-  candidate.time_candidate = data.timestamp;
-  // throw away bits 31-16 of header, that's OK for now
-  candidate.detid = { static_cast<triggeralgs::detid_t>(data.signal_map) }; // NOLINT(build/unsigned)
-  candidate.type = triggeralgs::TriggerCandidate::Type::kTiming;
 
-  candidate.algorithm = triggeralgs::TriggerCandidate::Algorithm::kHSIEventToTriggerCandidate;
-  candidate.inputs = {};
+    triggeralgs::TriggerCandidate candidate;
+    candidate.time_start = data.timestamp - m_hsisignal_map[data.signal_map].time_before;  // time_start
+    candidate.time_end   = data.timestamp + m_hsisignal_map[data.signal_map].time_after; // time_end,
+    candidate.time_candidate = data.timestamp;
+    // throw away bits 31-16 of header, that's OK for now
+    candidate.detid = { static_cast<triggeralgs::detid_t>(signal) }; // NOLINT(build/unsigned)
+
+    candidate.type = m_hsisignal_map[signal].type; // type,
+
+    candidate.algorithm = triggeralgs::TriggerCandidate::Algorithm::kHSIEventToTriggerCandidate;
+    candidate.inputs = {};
+    candidates.push_back(candidate);
+  }
 
-  return candidate;
+  return candidates;
 }
 
 
 void
 TimingTriggerCandidateMaker::do_conf(const nlohmann::json& config)
 {
   auto params = config.get<dunedaq::trigger::timingtriggercandidatemaker::Conf>();
-  m_detid_offsets_map[params.s0.signal_type] = { params.s0.time_before, params.s0.time_after };
-  m_detid_offsets_map[params.s1.signal_type] = { params.s1.time_before, params.s1.time_after };
-  m_detid_offsets_map[params.s2.signal_type] = { params.s2.time_before, params.s2.time_after };
-  m_hsi_passthrough = params.hsi_trigger_type_passthrough;
-  m_hsi_pt_before = params.s0.time_before;
-  m_hsi_pt_after = params.s0.time_after;
+
+  // Fill the internal hsi signal map. The map contains signal ID, TC type to
+  // create, and the readout time window.
+  for (auto hsi_input : params.hsi_configs) {
+    triggeralgs::TriggerCandidate::Type type;
+    type = static_cast<triggeralgs::TriggerCandidate::Type>(
+        dunedaq::trgdataformats::string_to_fragment_type_value(hsi_input.tc_type_name));
+    if (type == triggeralgs::TriggerCandidate::Type::kUnknown) {
+      throw ConfigurationProblem(ERS_HERE,
+          "Unknown TriggerCandidate supplied to TTCM HSI map");
+    }
+
+    if (m_hsisignal_map.count(hsi_input.signal)) {
+      throw ConfigurationProblem(ERS_HERE,
+          "Supplied more than one of the same hsi signal ID to TTCM HSI map");
+    }
+
+    m_hsisignal_map[hsi_input.signal] = { type,
+                                          hsi_input.time_before,
+                                          hsi_input.time_after };
+
+    TLOG() << "[TTCM] will convert HSI signal id: " << hsi_input.signal << " to TC type: " << hsi_input.tc_type_name;
```

Comment:
do we want this to be a TLOG instead of some low level debug ?

Reply:
I wanted all the "config" and "do stop" "do start" etc. to be simple TLOG(), because they will not be executed often. Can discuss further/change if needed though?

Reply:
The TLOG rework has set these 'one time messages' as GENERAL debug level, I think. Don't think it matters too much for these. I was mostly asking about the 'will convert HSI signal' message as that one is expected a lot more often.

Reply:
This message is only in the configuration, not in the conversion itself. 
So when receiving HSI signal, completely different lines of code will be triggered. 
This one will only be triggered once, when running `conf` through nanorc.